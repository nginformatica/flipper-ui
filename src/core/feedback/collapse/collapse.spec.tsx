import * as React from 'react'
import { render, screen } from '@testing-library/react'
import { Collapse } from '.'

describe('Collapse', () => {
    it('should render opened', () => {
        render(
            <Collapse in data-testid='collapse-container'>
                I am open for discussions
            </Collapse>
        )

        const label = screen.getByText('I am open for discussions')
        const container = screen.getByTestId('collapse-container')

        expect(container.classList).toContain('MuiCollapse-entered')
        expect(label).toBeDefined()
    })

    it('should render closed', () => {
        render(
            <Collapse data-testid='collapse-container' in={false}>
                I am not open for discussions
            </Collapse>
        )

        const container = screen.getByTestId('collapse-container')

        expect(container.classList).toContain('MuiCollapse-hidden')
    })
})
