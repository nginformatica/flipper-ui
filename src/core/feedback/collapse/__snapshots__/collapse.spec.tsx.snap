// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse should match snapshot 1`] = `
<div>
  <div
    class="MuiCollapse-root MuiCollapse-entered"
    data-testid="collapse-container"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner"
      >
        I am open for discussions
      </div>
    </div>
  </div>
</div>
`;
